Support another new Telegram version for Android

An user reported to me that important Telegram messages were not decoded (in fact just a few messages were actually decoded).
It seems that it is a recent database, still not supported by IPED's decoder.
I reproduced the issue using master and "telegram-decoder-impl-1.0.11_1.jar" here.
I updated TLRPC.java (getting the latest version from Telegram's project), fixed a few compiling issues that appeared (not sure how is the correct way of doing that), generated a new JAR and the messages were decoded properly using it.

![image](https://github.com/user-attachments/assets/f1187404-3cfb-413b-8008-d9ab35c23c28)

> I just update it to the latest available version in this branch https://github.com/sepinf-inc/telegram-decoder/tree/update_telegram_version. But I did not test it yet.

I tried to compile, but it fails:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /E:/git/telegram-decoder/telegram-decoder-impl/src/main/java/telegramdecoder/DecoderTelegram.java:[174,25] cannot find symbol
  symbol:   class var
  location: class telegramdecoder.DecoderTelegram
[ERROR] /E:/git/telegram-decoder/telegram-decoder-impl/src/main/java/telegramdecoder/DecoderTelegram.java:[175,51] cannot find symbol
  symbol:   class messageExtendedMediaPreview
  location: class org.telegram.tgnet.TLRPC
[ERROR] /E:/git/telegram-decoder/telegram-decoder-impl/src/main/java/telegramdecoder/DecoderTelegram.java:[176,34] cannot find symbol
  symbol:   class messageExtendedMediaPreview
  location: class org.telegram.tgnet.TLRPC
[ERROR] /E:/git/telegram-decoder/telegram-decoder-impl/src/main/java/telegramdecoder/DecoderTelegram.java:[176,79] cannot find symbol
  symbol:   class messageExtendedMediaPreview
  location: class org.telegram.tgnet.TLRPC
[INFO] 4 errors 
```

First error can be fixed by upgrading the project requirement to java 11 instead of 8, but others need additional fixes.

--- a/pom.xml
+++ b/pom.xml
@@ -43,7 +43,7 @@
         <java.dbx.version>1.1-p6</java.dbx.version>
         <xerial.sqlite.version>3.34.0</xerial.sqlite.version>
         <dockingframes.version>1.1.2</dockingframes.version>
-        <telegram.version>1.0.11_1</telegram.version>
+        <telegram.version>1.0.12</telegram.version>
         <twelvemonkeys.version>3.10.1</twelvemonkeys.version>
         <bencode.version>1.4.1</bencode.version>
     </properties>

